Updating the deploy-preview.yml

clarkbw opened the pull request:
Moving to use $GITHUB_ENV which should work for the migrate and deploy steps

Dropping some of the top level env variables.

Moved NEON_PROJECT_ID from secret to var

clarkbw committed:
Updating the deploy-preview.yml

Moving to use $GITHUB_ENV which should work for the migrate and deploy steps

Dropping some of the top level env variables.

Moved NEON_PROJECT_ID from secret to var

clarkbw commented on .github/workflows/deploy-preview.yml:
```diff
@@ -2,11 +2,9 @@ name: Preview Deployment
-  GH_TOKEN: ${{ secrets.GH_TOKEN }} # Required for commenting on pull requests for private repos
-  NEON_API_KEY: ${{ secrets.NEON_API_KEY }} # You can generate a an API key in your account settings
-  NEON_PROJECT_ID: ${{ secrets.NEON_PROJECT_ID }} # You can find this in your project settings
+  NEON_PROJECT_ID: ${{ var.NEON_PROJECT_ID }} # You can find this in your Neon project settings
```
This will require you moving the `NEON_PROJECT_ID` from secrets over to vars but it's not a protected secret so I think that's fine.

clarkbw replied:
If we move `NEON_PROJECT_ID` to vars it no longer needs to be listed up here. However I think it's good to have here so we can explain where it came from.

m-abdelwahab replied:
Yea I'm a fan of listing variables at the top of the workflow

clarkbw commented on .github/workflows/deploy-preview.yml:
```diff
@@ -21,46 +19,40 @@ jobs:
           version: 8
-      - name: Use Node.js 18
-        uses: actions/setup-node@v4
+      - uses: actions/setup-node@v4
```
Similar to code comments, I only like them when they are stating a reason that's not obvious. Otherwise they're superfluous.

m-abdelwahab replied:
Makes sense

clarkbw commented on .github/workflows/deploy-preview.yml:
```diff
@@ -21,46 +19,40 @@ jobs:
-      - name: Get branch name
+      - run: pnpm install
+
+      - name: Get cleaned up branch name
```
It wasn't clear to me why we needed this action, so naming it makes sense to me. Not sure this name is ideal though.

m-abdelwahab replied:
How about "Get git branch name"

clarkbw replied:
```suggestion
      - name: Get git branch name
```

clarkbw commented on .github/workflows/deploy-preview.yml:
```diff
@@ -21,46 +19,40 @@ jobs:
-      - name: Run Migrations
-        run: |
-          touch .env
+          username: ${{ secrets.NEON_DATABASE_USERNAME }} # change to your Neon database username / role
```
I don't think `NEON_DATABASE_USERNAME` is actually secret worthy but I don't think this matters too much.

m-abdelwahab replied:
Actually we didn't use a default value in the past, but now we default to `neondb_owner`, so might be worth removing this value

clarkbw replied:
Yes, let's do that.
```suggestion
```

clarkbw commented on .github/workflows/deploy-preview.yml:
```diff
@@ -21,46 +19,40 @@ jobs:
-        run: |
-          touch .env
+          username: ${{ secrets.NEON_DATABASE_USERNAME }} # change to your Neon database username / role
+          api_key: ${{ secrets.NEON_API_KEY }} # Generate a an API key in your Neon account settings
```
Moved this away from the `env` header because otherwise all the previous steps have access to this secret. Ideally we only expose it to the actions who need it.

m-abdelwahab replied:
Makes sense

clarkbw commented on .github/workflows/deploy-preview.yml:
```diff
@@ -21,46 +19,40 @@ jobs:
 
-          echo DATABASE_URL=${{ steps.create-branch.outputs.db_url_with_pooler }}?sslmode=require >> .env
+      - run: |
+          echo "DATABASE_URL=${{ steps.create-branch.outputs.db_url_with_pooler }}?sslmode=require" >> "$GITHUB_ENV"
```
This feels cleaner to me compared to creating the `.env` file. And it should work the same way... testing needed.

clarkbw commented on .github/workflows/deploy-preview.yml:
```diff
@@ -21,46 +19,40 @@ jobs:
         uses: superfly/fly-pr-review-apps@1.2.0
         with:
-          secrets: DATABASE_URL=${{ steps.create-branch.outputs.db_url }}?sslmode=require
+          secrets: DATABASE_URL=$DATABASE_URL
```
This should be picking up the env variable created with the `>> $GITHUB_ENV` step.

clarkbw commented on .github/workflows/deploy-preview.yml:
```diff
@@ -21,46 +19,40 @@ jobs:
           message: |
             Fly Preview URL :balloon: : ${{ steps.deploy.outputs.url }}
-            Neon branch :elephant: : https://console.neon.tech/app/projects/${{ secrets.NEON_PROJECT_ID }}/branches/${{ steps.create-branch.outputs.branch_id }}
+            Neon branch :elephant: : https://console.neon.tech/app/projects/${{ env.NEON_PROJECT_ID }}/branches/${{ steps.create-branch.outputs.branch_id }}
```
Pulls from the env variable at the top of the file instead of the secret directly.

clarkbw commented:
Looks good to me!

m-abdelwahab committed:
Add name for the get git branch name step

Co-authored-by: Bryan Clark <clarkbw@gmail.com>

clarkbw committed:
Update cleanup-preview.yml

Use the var.NEON_PROJECT_ID to match

clarkbw committed:
Update deploy-production.yml

Truncate step names
Move env variables for this one down to the steps that require them. We've explained in other files where they are needed.

m-abdelwahab committed:
Remove database username

Co-authored-by: Bryan Clark <clarkbw@gmail.com>
